charmcraft/config.py

```diff
@@ -142,6 +219,24 @@ class Config:
 
 CONFIG_SCHEMA = {
     'type': 'object',
+    'definitions': {
+        'base': {
+            'type': 'object',
+            'properties': {
+                'name': {'type': 'string'},
+                'channel': {"oneOf": [{'type': 'string'}, {'type': 'number'}]},
```

Comment:
IMO channel must be always a string, not a number. Better explicit than implicit.